src/EnergyPlus/IndoorGreen.cc

```diff
@@ -0,0 +1,778 @@
+                lw->indoorgreen(IndoorGreenNum).SensibleRate =
+                    (1-lw->indoorgreen(IndoorGreenNum).LEDRadFraction) * lw->indoorgreen(IndoorGreenNum).LEDActualEleP; // convective heat gain from LED lights when LED is on; heat convection from plants was considered and counted from plant surface heat balance.
+                lw->indoorgreen(IndoorGreenNum).LatentRate = ZoneAirVol * rhoair * (HCons - HMid) / Timestep;   // unit W
+                state.dataHeatBalSurf->SurfQAdditionalHeatSourceInside(lw->indoorgreen(IndoorGreenNum).SurfPtr) =
+                    -1.0*lw->indoorgreen(IndoorGreenNum).LambdaET;
+            }
+    }
+
+
+    Real64 ETPenmanMonteith(EnergyPlusData &state, Real64 &ZonePreTemp, Real64 &ZonePreHum, Real64 &ZoneCO2, Real64 &ZonePPFD, Real64 &ZoneVPD, Real64 &LAI)
+    {
+            // PURPOSE OF THIS SUBROUTINE:
+            // This subroutine is for using Penman-Monteith ET model to calculate evapotranspiration rates from the Indoor Greenery System objects.
+            // Reference: Monteith, J.L. Evaporation and environment. in Symposia of the society for experimental biology. 1965. Cambridge University Press (CUP) Cambridge
+            
+            // SUBROUTINE PARAMETER DEFINITIONS:
+            static constexpr std::string_view RoutineName("ETPenmanMonteith: ");
+            Real64 hfg = Psychrometrics::PsyHfgAirFnWTdb(ZonePreHum, ZonePreTemp)/std::pow(10,6); // Latent heat of vaporization (MJ/kg)
+            Real64 slopepat = 0.200 * std::pow((0.00738 * ZonePreTemp + 0.8072), 7) - 0.000116; //Slope of the saturation vapor pressure-temperature curve (kPa/°C)
+            Real64 CpAir = Psychrometrics::PsyCpAirFnW(ZonePreHum) / std::pow(10, 6); // specific heat of air at constant pressure (MJ kg−1 °C−1)
+            Real64 OutPb = state.dataEnvrn->OutBaroPress / 1000;    // outdoor pressure (kPa)
+            Real64 constexpr mw(0.622);//ratio molecular weight of water vapor / dry air = 0.622.
+            Real64 psyconst = CpAir * OutPb / (hfg * mw); // Psychrometric constant (kPa/°C)
+            Real64 rs =0.0; //stomatal resistance s/m
+            Real64 ra =0.0; //aerodynamic resistance s/m
+            Real64 In = ZonePPFD * 0.327 / std::pow(10, 6); // net radiation MW/m2
+            Real64 G = 0.0; //soil heat flux (MJ/(m2s))
+            Real64 rhoair = Psychrometrics::PsyRhoAirFnPbTdbW(state, OutPb * 1000, ZonePreTemp, ZonePreHum); //kg/m3
+            Real64 ETRate; //mm/s; kg/(m2s)
+            rs =60*(1500+ZonePPFD)/(200+ZonePPFD);
+            ra =350*std::pow((0.1/0.1),0.5)*(1/(LAI+1e-10));
+            ETRate = (1 / hfg) * (slopepat * (In - G) + (rhoair * CpAir * ZoneVPD) / ra) / (slopepat + psyconst * (1 + rs / ra));//Penman-Monteith ET model
+            return ETRate; //mm/s; kg/(m2s)
+    }
+    Real64
+    ETStanghellini(EnergyPlusData &state, Real64 &ZonePreTemp, Real64 &ZonePreHum, Real64 &ZoneCO2, Real64 &ZonePPFD, Real64 &ZoneVPD, Real64 &LAI)
```

Comment:
Except for `state`, these are all scalar inputs. They should be passed in by value, not reference.

Reply:
Thank you! Removed reference & in ET model inputs.